protocol/streaming/full_node_streaming_manager.go

```diff
@@ -501,28 +483,37 @@ func (sm *FullNodeStreamingManagerImpl) TracksSubaccountId(subaccountId satypes.
 }
 
 func getStreamUpdatesFromOffchainUpdates(
-	offchainUpdates *clobtypes.OffchainUpdates,
+	v1updates []ocutypes.OffChainUpdateV1,
 	blockHeight uint32,
 	execMode sdk.ExecMode,
 ) (streamUpdates []clobtypes.StreamUpdate, clobPairIds []uint32) {
 	// Group updates by clob pair id.
-	updates := make(map[uint32]*clobtypes.OffchainUpdates)
-	for _, message := range offchainUpdates.Messages {
-		clobPairId := message.OrderId.ClobPairId
-		if _, ok := updates[clobPairId]; !ok {
-			updates[clobPairId] = clobtypes.NewOffchainUpdates()
+	clobPairIdToV1Updates := make(map[uint32][]ocutypes.OffChainUpdateV1)
+	for _, v1update := range v1updates {
+		var clobPairId uint32
+		switch u := v1update.UpdateMessage.(type) {
+		case *ocutypes.OffChainUpdateV1_OrderPlace:
+			clobPairId = u.OrderPlace.Order.OrderId.ClobPairId
+		case *ocutypes.OffChainUpdateV1_OrderReplace:
+			clobPairId = u.OrderReplace.OldOrderId.ClobPairId
+		case *ocutypes.OffChainUpdateV1_OrderRemove:
+			clobPairId = u.OrderRemove.RemovedOrderId.ClobPairId
+		case *ocutypes.OffChainUpdateV1_OrderUpdate:
+			clobPairId = u.OrderUpdate.OrderId.ClobPairId
+		default:
+			panic(fmt.Sprintf("Unhandled UpdateMessage type: %v", u))
 		}
-		updates[clobPairId].Messages = append(updates[clobPairId].Messages, message)
+
+		if _, ok := clobPairIdToV1Updates[clobPairId]; !ok {
+			clobPairIdToV1Updates[clobPairId] = []ocutypes.OffChainUpdateV1{}
+		}
+		clobPairIdToV1Updates[clobPairId] = append(clobPairIdToV1Updates[clobPairId], v1update)
 	}
 
 	// Unmarshal each per-clob pair message to v1 updates.
 	streamUpdates = make([]clobtypes.StreamUpdate, 0)
 	clobPairIds = make([]uint32, 0)
-	for clobPairId, update := range updates {
-		v1updates, err := streaming_util.GetOffchainUpdatesV1(update)
-		if err != nil {
-			panic(err)
-		}
+	for clobPairId, v1updates := range clobPairIdToV1Updates {
```

Comment:
can we iterate over maps deterministically here?

Reply:
Done